pkgs/development/libraries/optparse-bash/default.nix

```
@@ -0,0 +1,39 @@
+{ stdenvNoCC
+, lib
+, fetchFromGitHub
+, gnused
+}:
+let
+  pname = "optparse";
+  version = "unstable-20170611";
+in stdenvNoCC.mkDerivation {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "nk412";
+    repo = pname;
+    rev = "0dfc5a3344feafc818b1fb051e4bebd195d18893";
+    sha256 = "sha256-fFDKU4ZQxrDokxwjHpXQaL3XGwdrSPugQSJbXMi6tJY=";
+  };
+
+  propagatedBuildInputs = [ gnused ];
+
+  outputs = [ "out" "lib" ];
+
+  installPhase = ''
+    mkdir -p $lib/libexec
+    mv optparse.bash $lib/libexec/
+    mkdir -p $out/bin
+    ln -s $lib/libexec/optparse.bash $out/bin/optparse.bash
```

```suggestion
    ln -s $lib/libexec/optparse.bash $out/bin/optparse
```


file in bin have usually no extension.

---

`optparse.bash` is meant to be `source`d in bash scripts instead of executed.

The .bash extension appears in the project README and the example script, so it would still be needed.